service/storage_service.hh

```diff
@@ -326,6 +335,10 @@ private:
      * 1. this node is being upgraded from a non-CDC version,
      * 2. this node is starting for the first time or restarting with CDC previously disabled,
      *    in which case the value should become populated before we leave the join_token_ring procedure.
+     *
+     * Important: this variable is using only during the startup procedure. It is moved out from
+     * at the end of join_token_ring; the responsibility handling of CDC generations is passed
+     * to cdc::generation_service.
```

Comment:
From [cppreference](https://en.cppreference.com/w/cpp/utility/optional/optional) (emphasis mine):

> 3) Move constructor: If other contains a value, initializes the contained value as if direct-initializing (but not direct-list-initializing) an object of type T with the expression std::move(*other) and __does not make other empty__: a moved-from optional still contains a value, but the value itself is moved from. If other does not contain a value, constructs an object that does not contain a value.

Nevertheless, even if the behavior described here was true, I think moving `std::optional` and expecting the source value to become `nullopt` is not clear. Please explicitly clear `_cdc_streams_ts` after passing it to `after_join`.

Reply:
I'm not expecting it to become `nullopt`, I'm expecting it to not be used by any one

Reply:
also the comment should say "used" not "using"

Reply:
A better solution would probably be to remove this field from storage_service altogether... but then I'd have to plumb it around through all these startup functions: prepare_to_join, init_server, join_token_ring...

Actually, I like this idea, if we broaden it to all state that is used during node startup/bootstrap.
That is, we could have a `startup_state` struct that is passed in through the startup code. The startup code would consist of free functions that would be called by `main`, and those functions would then return a result that would be used to initialize other services.
There is really no reason for the startup procedure to consist of `storage_service`'s methods.

Reply:
Ok, I see. Thanks for the clarification.

Reply:
I should add a line in this comment saying explicitly "do not ever use this variable after join_token_ring" or sth